packages/mobile/src/screens/edit-track-screen/screens/TrackAvailabilityScreen.tsx

```diff
@@ -104,27 +105,37 @@ export const TrackAvailabilityScreen = () => {
     // eslint-disable-next-line
   }, [])
 
-  // we only care about what the initial value was here
-  // eslint-disable-next-line
-  const initialPremiumConditions = useMemo(() => premiumConditions, [])
+  const isInitiallyPublic =
+    !isUpload && !initialValues.is_unlisted && !initialPremiumConditions
 
   const isInitiallySpecialAccess =
-    !isUpload && initialAvailability === TrackAvailabilityType.SPECIAL_ACCESS
+    !isUpload &&
+    !!(
+      initialPremiumConditions?.follow_user_id ||
+      initialPremiumConditions?.tip_user_id
+    )
   const noCollectibleGate =
-    isInitiallySpecialAccess || isRemix || hasNoCollectibles
+    isInitiallyPublic ||
+    isInitiallySpecialAccess ||
+    isRemix ||
+    hasNoCollectibles
   const noCollectibleDropdown = isInitiallySpecialAccess || !isUpload
 
   const isInitiallyCollectibleGated =
-    !isUpload && initialAvailability === TrackAvailabilityType.COLLECTIBLE_GATED
-  const noSpecialAccess = isInitiallyCollectibleGated || isRemix
+    !isUpload && !!initialPremiumConditions?.nft_collection
+  const noSpecialAccess =
+    isInitiallyPublic || isInitiallyCollectibleGated || isRemix
   const noSpecialAccessOptions = noSpecialAccess || !isUpload
 
-  const noHidden =
-    !isUpload && initialAvailability !== TrackAvailabilityType.HIDDEN
+  const noHidden = !isUpload && !initialValues.is_unlisted
 
   const [availability, setAvailability] =
     useState<TrackAvailabilityType>(initialAvailability)
 
+  // we only care about what the initial value was here
+  // eslint-disable-next-line
```

Comment:
nit: we should disable just the rule we care about to prevent this from hiding other errors